src/client/driver/command-executors/client-functions/selector-executor/index.js

```diff
@@ -109,4 +116,28 @@ export default class SelectorExecutor extends ClientFunctionExecutor {
             })
             .then(() => element);
     }
+
+    getResultDriverStatus () {
+        return this
+            .getResult()
+            .then(result => {
+                var node               = result;
+                var snapshot           = this.replicator.encode(result)[0];
```

Comment:
We need to pass extensions to replicator and appropriate transformers somehow. If you attach them afterwards, newly added properties will not be processed by replicator and thus circular references will throw errors and you will not be able to encode advanced data types. Also, note that selector can work in `counter` and `collection` modes: https://github.com/kirovboris/testcafe-phoenix/blob/5e34753e68f732c0ad639f9076c349330422014f/src/client/driver/command-executors/client-functions/selector-executor/filter.js#L101 First one used to calculate `.exists` and `.count` and the second one is used in chains. Make sure everything works correctly with extensions in this mode.